app/src/main/java/net/squanchy/venue/VenueInfoPageView.java

```diff
@@ -0,0 +1,123 @@
+package net.squanchy.venue;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.ContextWrapper;
+import android.support.v7.widget.Toolbar;
+import android.util.AttributeSet;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import net.squanchy.R;
+import net.squanchy.imageloader.ImageLoader;
+import net.squanchy.imageloader.ImageLoaderInjector;
+import net.squanchy.navigation.LifecycleView;
+import net.squanchy.navigation.Navigator;
+import net.squanchy.venue.domain.view.Venue;
+
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.Disposable;
+
+public class VenueInfoPageView extends LinearLayout implements LifecycleView {
+
+    private Disposable subscription;
+    private Navigator navigate;
+    private VenueInfoService service;
+    private TextView nameText;
+    private TextView addressText;
+    private TextView descriptionText;
+    private ImageView mapView;
+    private ImageLoader imageLoader;
+
+    public VenueInfoPageView(Context context, AttributeSet attrs) {
+        this(context, attrs, 0);
+    }
+
+    public VenueInfoPageView(Context context, AttributeSet attrs, int defStyleAttr) {
+        this(context, attrs, defStyleAttr, 0);
+    }
+
+    public VenueInfoPageView(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
+        super(context, attrs, defStyleAttr, defStyleRes);
+
+        super.setOrientation(VERTICAL);
+    }
+
+    @Override
+    public void setOrientation(int orientation) {
+        throw new UnsupportedOperationException("TweetsPageView doesn't support changing orientation");
+    }
+
+    @Override
+    protected void onFinishInflate() {
+        super.onFinishInflate();
+
+        Activity activity = unwrapToActivityContext(getContext());
+        VenueInfoComponent component = VenueInfoInjector.obtain(activity);
+        navigate = component.navigator();
+        service = component.service();
+        imageLoader = ImageLoaderInjector.obtain(getContext()).imageLoader();
+
+        nameText = (TextView) findViewById(R.id.venue_name);
+        addressText = (TextView) findViewById(R.id.venue_address);
+        descriptionText = (TextView) findViewById(R.id.venue_description);
+        mapView = (ImageView) findViewById(R.id.venue_map);
+
+        setupToolbar();
+    }
+
+    private static Activity unwrapToActivityContext(Context context) {
```

Comment:
I suppose we could extract this if we use it in several places now, to a helper class or something.